### PR TITLE
scripts/cargo.sh: document variables in scripts/cargo.sh

### PR DESCRIPTION
**Pull request by maxice8**

### CODE REVIEW

File: scripts/cargo.sh [hunk 1]
```diff
@@ -1,14 +1,27 @@
 #!/bin/sh
 
+# $1 -> @CURRENT_SOURCE_DIR@ as replaced by meson, we cd into it
```
Comment (Cogitri) [inline scripts/cargo.sh]: This isn't current source dir for the xi-syntect-plugin

File: scripts/cargo.sh [hunk 2]
```diff
@@ -1,14 +1,27 @@
...
+# $2 -> Where to place the finished binaries, meson build_root with @OUTDIR@
```
Comment (Cogitri) [inline scripts/cargo.sh]: There's a point missing between meson and build_root

File: scripts/cargo.sh (hunk 1 above)
Comment (maxice8) [reply scripts/cargo.sh]: can't process this one

File: scripts/cargo.sh (hunk 2 above)
Comment (maxice8) [reply scripts/cargo.sh]: fixed

File: scripts/cargo.sh (hunk 1 above)
Comment (Cogitri) [reply scripts/cargo.sh]: In meson.build we have two buils targets which invoke this: `cargo-build-gxi` and `cargo-build-syntect`. In the former your comment holds true, but in the latter case we use `syntect_build_path`

**Comment by Cogitri** [comment]
Seems like CI isn't happy about the change

**Comment by maxice8** [comment]
On Mon, Apr 01, 2019 at 01:18:32PM -0700, Rasmus Thomsen wrote:
> Seems like CI isn't happy about the change
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/Cogitri/gxi/pull/165#issuecomment-478731266

:c

File: scripts/cargo.sh [hunk 3]
```diff
@@ -18,19 +31,20 @@ echo "\tGXI Plugindir: ${GREEN}${GXI_PLUGIN_DIR}${NO_COLOR}
...
+
 if [ "$7" = "true" ]; then
     features="--features gtk_v3_22"
 fi
 
 if [ "$GXI_BUILD_PROFILE" = "development" ]; then
-	path=$(echo $path | sed 's|release|debug|')
-	cargo build --target-dir ${4} ${features} && cp "${path}" ${2}/${3}
+	path="$(echo $path | sed 's|release|debug|')"
+	cargo build --target-dir "${4}" "${features}" && cp "${path}" "${2}/${3}"
```
Comment (Cogitri) [inline scripts/cargo.sh]: The quotes from features have to go for this to work, otherwise it interprets the '--features gtk_v3_22' as one argument.

Comment (maxice8) [reply scripts/cargo.sh]: done

**Pull request merged**